### PR TITLE
#163426617  Integrate unit testing with Travis ci

### PR DESCRIPTION
**Pull request by mrkingz**
#### What does this PR do?
 - Integrate unit testing with Travis ci 
 #### Description of Task to be completed?
 - To ensure the back end has all functionality working as intended
 #### What are the relevant pivotal tracker stories?
- [#163426617](https://www.pivotaltracker.com/n/projects/2240406/stories/163426617)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,12 +30,18 @@
       "git add"
     ]
   },
+  "nyc": {
+    "exclude": [
+      "dist/index.js",
```
Comment (benfluleck) [inline package.json]: why not exclude the dist folder

File: package.json [hunk 2]
```diff
@@ -66,15 +74,23 @@
...
+    "mocha": "^5.2.0",
     "mocha-lcov-reporter": "^1.3.0",
-    "nodemon": "^1.18.9",
     "nyc": "^13.1.0",
     "prettier-cli": "^0.1.0",
     "prettier-eslint": "^8.8.2",
-    "prettier-eslint-cli": "^4.7.1"
+    "prettier-eslint-cli": "^4.7.1",
+    "sqlite3": "^4.0.6",
+    "supertest": "^3.4.2"
```
Comment (benfluleck) [inline package.json]: Why do we have supertest and chai. Please remove one of them.

File: package.json [hunk 3]
```diff
@@ -66,15 +74,23 @@
...
     "coveralls": "^3.0.2",
+    "cross-env": "^5.2.0",
+    "dotenv": "^6.2.0",
     "eslint": "^5.13.0",
+    "eslint-config-airbnb-base": "^13.1.0",
+    "eslint-plugin-import": "^2.16.0",
     "husky": "^1.3.1",
     "lint-staged": "^8.1.1",
+    "mocha": "^5.2.0",
     "mocha-lcov-reporter": "^1.3.0",
```
Comment (benfluleck) [inline package.json]: What is this doing in here please uninstall

File: server/tests/miscellaneous/connection.spec.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import chai from "chai";
+import chaiHttp from "chai-http";
+import app from "../../index";
+
+chai.use(chaiHttp);
+const expect = chai.expect;
+
+describe("Server", () => {
```
Comment (benfluleck) [inline server/tests/miscellaneous/connection.spec.js]: `describe("GET <API />",`

File: server/tests/miscellaneous/connection.spec.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  it("should return Connection ok", done => {
```
Comment (benfluleck) [inline server/tests/miscellaneous/connection.spec.js]: This is not descriptive enough. The test should tell me exactly what it is doing?

`it("should return successful connection")` or ` it("should return 200 status code") `

File: server/tests/miscellaneous/connection.spec.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    chai
+      .request(app)
+      .post("/api")
+      .end((err, res) => {
+        expect(res.statusCode).to.equal(200);
+        expect(res.body.status).to.equal("Success");
+        expect(res.body.message).to.equal("Connection ok");
+        done();
+      });
+  });
```
Comment (benfluleck) [inline server/tests/miscellaneous/connection.spec.js:19]: space between tests

File: server/tests/miscellaneous/connection.spec.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  it("should return a 404 error", done => {
+    chai
+      .request(app)
+      .post("/api/blahblahblah")
+      .end((err, res) => {
+        expect(res.statusCode).to.equal(404);
+        done();
+      });
+  });
```
Comment (benfluleck) [inline server/tests/miscellaneous/connection.spec.js:29]: space between tests

File: server/index.js [hunk 8]
```diff
@@ -1,9 +1,11 @@
 import express from "express";
 import session from "express-session";
 import cors from "cors";
+import dotenv from "dotenv";
 import errorhandler from "errorhandler";
```
Comment (benfluleck) [inline server/index.js]: What is this package doing here please remove it

**Review by benfluleck — CHANGES_REQUESTED** [review]

File: server/index.js [hunk 9]
```diff
@@ -69,16 +76,24 @@ app.use((err, req, res, next) => {
...
+    status: "Success",
+    message: "Connection ok"
+  });
+});
+
 // finally, let's start our server...
 db.sequelize
   .authenticate()
-  .then(() => {
+  .then(re => {
```
Comment (benfluleck) [inline server/index.js]: What is this `re`?

File: server/index.js [hunk 10]
```diff
@@ -44,23 +40,27 @@ app.get("/", (req, res, next) => {
...
 });
-/// catch 404 and forward to error handler
+
+// catch 404 and forward to error handler
 app.use((error, req, res, next) => {
   if (error) {
     const err = new Error("Not Found");
-    err.status = 404;
-    return next(err);
+    err.status = 500;
```
Comment (benfluleck) [inline server/index.js]: Why did you change this to 500

File: package.json [hunk 11]
```diff
@@ -66,15 +64,21 @@
     "@babel/plugin-proposal-object-rest-spread": "^7.3.1",
     "@babel/plugin-transform-runtime": "^7.2.0",
     "@babel/preset-env": "^7.3.1",
+    "chai": "^4.2.0",
+    "chai-http": "^4.2.1",
     "coveralls": "^3.0.2",
+    "cross-env": "^5.2.0",
+    "dotenv": "^6.2.0",
```
Comment (Ndifreke) [inline package.json]: Don't forget to take this out as requested by @benfluleck

File: package.json [hunk 12]
```diff
@@ -66,15 +64,21 @@
...
     "eslint": "^5.13.0",
     "husky": "^1.3.1",
     "lint-staged": "^8.1.1",
+    "mocha": "^5.2.0",
     "mocha-lcov-reporter": "^1.3.0",
```
Comment (benfluleck) [inline package.json]: Why is this still here ?

**Review by eokwukwe — COMMENTED** [review]
Kingsley, I thought you were told to remove files and codes that we don't need. I can still see `mongoose` in `server/configs/config.js`

**Review by obasajujoshua31 — COMMENTED** [review]
This is ok

**Review by Ndifreke — APPROVED** [review]
I thinks this is ready to go @benfluleck

**Pull request merged**